File: src/test/java/pl/training/module5/Module5ExamplesTest.java
package pl.training.module5;

import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;

import org.junit.jupiter.api.Test;

final class Module5ExamplesTest {
    @Test
    void runsAllModuleExamples() {
        assertDoesNotThrow(() -> Module5Examples.main(new String[0]));
    }
}

File: csharp/tests/Training.Module5.Tests/Module5ExamplesTest.cs
namespace Training.Module5.Tests;

[Collection(GlobalStateCollection.Name)]
public sealed class Module5ExamplesTest
{
    [Fact]
    public void RunsAllModuleExamples()
    {
        var originalOut = Console.Out;
        using var output = new StringWriter();
        try
        {
            Console.SetOut(output);

            var exception = Record.Exception(() => Module5Examples.Main([]));

            Assert.Null(exception);
        }
        finally
        {
            Console.SetOut(originalOut);
        }

        var nl = Environment.NewLine;
        Assert.Equal(
            "Hierarchy stages equivalent: true" + nl
            + "Extract subclass equivalent: true" + nl
            + "Collapse hierarchy equivalent: true" + nl
            + "Composition client behavior equivalent: true" + nl
            + "Batch results: [mail-1|OPS|Ready|EMAIL|SENT, sms-1|OPS|Ready|SMS|SENT|RECEIPT]" + nl,
            output.ToString());
    }
}
